src/modulefinder/sentry_modulefinder_linux.c

```diff
@@ -419,13 +475,53 @@ sentry__procmaps_module_to_value(const sentry_module_t *module)
     const sentry_mapped_region_t *first_mapping = &module->mappings[0];
     const sentry_mapped_region_t *last_mapping
         = &module->mappings[module->num_mappings - 1];
+    uint64_t module_size
+        = last_mapping->addr + last_mapping->size - first_mapping->addr;
+
     sentry_value_set_by_key(
         mod_val, "image_addr", sentry__value_new_addr(first_mapping->addr));
-    sentry_value_set_by_key(mod_val, "image_size",
-        sentry_value_new_int32(
-            last_mapping->addr + last_mapping->size - first_mapping->addr));
+    sentry_value_set_by_key(
+        mod_val, "image_size", sentry_value_new_int32(module_size));
+
+    // At least on the android API-16, x86 simulator, the linker apparently
+    // does not load the complete file into memory. Or at least, the section
+    // headers which are located at the end of the file are not loaded, and
+    // we would be poking into invalid memory. To be safe, we mmap the
+    // complete file from disk, so we have the on-disk layout, and are
+    // independent of how the runtime linker would load or re-order any
+    // sections. The exception here is the linux-gate, which is not an
+    // actual file on disk, so we actually poke at its memory.
+    if (sentry__slice_eq(module->file, LINUX_GATE)) {
```

Comment:
pretty sure its more than just gate, they're all known as "vdso" and they appear under several names, see https://man7.org/linux/man-pages/man7/vdso.7.html which includes a table.

Reply:
Its still only a single one as the man page says. They just have different names depending on the platform. However we synthesize the name anyway, so I don’t think that matters much.